torch/_inductor/autotune_process.py

```diff
@@ -34,58 +34,45 @@ class Pong:
 
 @dataclasses.dataclass
 class TuningProcess:
-    process: Optional[BaseProcess] = None
-    request_queue: Optional["Queue[Any]"] = None
-    response_queue: Optional["Queue[Any]"] = None
+    process: Optional["subprocess.Popen[bytes]"] = None
 
     @staticmethod
-    def process_main(
-        request_queue: "Queue[Any]",
-        response_queue: "Queue[Any]",
-    ) -> None:
-        print("enter child process main")
-        while True:
-            obj = request_queue.get()
+    def process_main() -> None:
+        """
+        Work loop for the benchmarking subprocess.
+        """
+        if DEBUG:
+            print("Entering TuningProcess child main", file=sys.stderr)
```

Comment:
Just add this as a logger call instead of a temporary print statement.

Reply:
@Skylion007, yeah these print stmts were already here and I just changed them to dump to stderr since this refactor uses stdout for the IPC. So printing to stdout would mess up the messaging. Arguably these prints were never going to be helpful in the first place, however, because the sub-process doesn't have a terminal to print them. I can just remove them since I'm not sure any of the 3 instances are terribly useful ... Or, the "right" approach is probably to dump to a file. If you recommend that approach, do you know if this code base has a canonical pattern for writing debug info to a tmp file?

Reply:
ellison has a similar comment; I'll adding logging to a file

Reply:
My bad using a DEBUG flag in the first place. Back then we don't have TORCH_LOGS support ..

Does using logging.Logger.debug work here? Those message should by default go to stderr

Reply:
@shunting314, no problem. I'm happy to change it. Just wasn't sure how much to evolve the implementation in one PR. The problem with logging normally to stderr is that this code runs in the sub-process, and there's no console. I was gonna take a crack at just logging to a file instead. I think that might be the most usable especially after the next diff in the stack where we have multiple independent sub-processes. What do you think?

Reply:
hmm, didn't subprocess inherits stderr from parent process? Logging to a separate file at lease makes it hard to check output in senarios like CI. And people need to be aware there is a separate file to check. If stderr works, I'll prefer stderr.


And yea, this can be done in separate PR

Reply:
> hmm, didn't subprocess inherits stderr from parent process?

I couldn't get it to work out of the box, but as Elias might tell you, I'm kind of a noob working with python sub-processes ;)  I'll look again and figure it out.